staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -718,80 +719,131 @@ func diffListsOfMaps(original, modified []interface{}, schema LookupPatchMeta, m
 	patch := make([]interface{}, 0, len(modified))
 	deletionList := make([]interface{}, 0, len(original))
 
-	originalSorted, err := sortMergeListsByNameArray(original, schema, mergeKey, false)
+	originalKeys, originalMap, err := mergeListToMapAndKeys(original, schema, mergeKey)
 	if err != nil {
 		return nil, nil, err
 	}
-	modifiedSorted, err := sortMergeListsByNameArray(modified, schema, mergeKey, false)
+	modifiedKeys, modifiedMap, err := mergeListToMapAndKeys(modified, schema, mergeKey)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	originalIndex, modifiedIndex := 0, 0
```

Comment:
Does this change how the client generates a patch to send to the server, or how the server handles an existing patch request from the client?

Reply:
> This method changes the way how the client generates patch command and PR in general affects both client and server side.

Then we have to reason about the following combinations:
1. new client, new server. This is the only case we're exercising in unit tests now, I think (since both the client and server side are running with new code).
2. new client, old server. We have to capture the change in what a new client would generate/send as a patch, and find a way to demonstrate that an old server will do something no worse than what it does today.
3. old client, new server. We have to capture what an old client would generate/send as a patch, and make sure a new server will produce the same result when given that patch (which is one reason the ordering changes to the existing unit test were concerning).